// @strapi/design-system/dist/SubNav/SubNavLink.cjs
"use strict";var $=Object.defineProperty;var n=Object.getOwnPropertySymbols;var f=Object.prototype.hasOwnProperty,y=Object.prototype.propertyIsEnumerable;var p=(e,t,l)=>t in e?$(e,t,{enumerable:!0,configurable:!0,writable:!0,value:l}):e[t]=l,g=(e,t)=>{for(var l in t||(t={}))f.call(t,l)&&p(e,l,t[l]);if(n)for(var l of n(t))y.call(t,l)&&p(e,l,t[l]);return e};var m=(e,t)=>{var l={};for(var o in e)f.call(e,o)&&t.indexOf(o)<0&&(l[o]=e[o]);if(e!=null&&n)for(var o of n(e))t.indexOf(o)<0&&y.call(e,o)&&(l[o]=e[o]);return l};Object.defineProperties(exports,{__esModule:{value:!0},[Symbol.toStringTag]:{value:"Module"}});const _=require("react"),B=require("prop-types"),S=require("styled-components"),T=require("@strapi/icons/Dot"),k=require("react-router-dom"),s=require("../Box/Box.cjs"),x=require("../Typography/Typography.cjs"),b=require("../Flex/Flex.cjs"),u=e=>e&&typeof e=="object"&&"default"in e?e:{default:e},r=u(_),a=u(B),c=u(S),q=u(T),E=c.default(s.Box)`
  display: flex;
  align-items: center;
  justify-content: space-between;
  text-decoration: none;
  color: ${({theme:e})=>e.colors.neutral800};
  svg > * {
    fill: ${({theme:e})=>e.colors.neutral600};
  }

  &.active {
    ${({theme:e})=>`
      background-color: ${e.colors.primary100};
      border-right: 2px solid ${e.colors.primary600};
      svg > * {
        fill: ${e.colors.primary700};
      }
      ${x.Typography} {
        color: ${e.colors.primary700};
        font-weight: 500;
      }
      `}
  }

  &:focus-visible {
    outline-offset: -2px;
  }
`,h=c.default(q.default)`
  width: ${12/16}rem;
  height: ${4/16}rem;
  * {
    fill: ${({theme:e,$active:t})=>t?e.colors.primary600:e.colors.neutral600};
  }
`,L=c.default.div`
  svg {
    height: ${12/16}rem;
    width: ${12/16}rem;
  }
`,i=j=>{var d=j,{children:e,icon:t,withBullet:l,isSubSectionChild:o}=d,v=m(d,["children","icon","withBullet","isSubSectionChild"]);return r.default.createElement(E,g({as:k.NavLink,icon:t,background:"neutral100",paddingLeft:o?9:7,paddingBottom:2,paddingTop:2},v),r.default.createElement(b.Flex,null,t?r.default.createElement(L,null,t):r.default.createElement(h,null),r.default.createElement(s.Box,{paddingLeft:2},r.default.createElement(x.Typography,{as:"span"},e))),l&&r.default.createElement(s.Box,{as:b.Flex,paddingRight:4},r.default.createElement(h,{$active:!0})))};i.displayName="SubNavLink";i.defaultProps={icon:null,isSubSectionChild:!1,withBullet:!1};i.propTypes={children:a.default.node,icon:a.default.element,isSubSectionChild:a.default.bool,link:a.default.element,withBullet:a.default.bool};exports.SubNavLink=i;
