// @strapi/design-system/dist/SubNav/SubNavLink.js
var v = Object.defineProperty;
var n = Object.getOwnPropertySymbols;
var d = Object.prototype.hasOwnProperty, u = Object.prototype.propertyIsEnumerable;
var m = (e, r, o) => r in e ? v(e, r, { enumerable: !0, configurable: !0, writable: !0, value: o }) : e[r] = o, f = (e, r) => {
  for (var o in r || (r = {}))
    d.call(r, o) && m(e, o, r[o]);
  if (n)
    for (var o of n(r))
      u.call(r, o) && m(e, o, r[o]);
  return e;
};
var g = (e, r) => {
  var o = {};
  for (var t in e)
    d.call(e, t) && r.indexOf(t) < 0 && (o[t] = e[t]);
  if (e != null && n)
    for (var t of n(e))
      r.indexOf(t) < 0 && u.call(e, t) && (o[t] = e[t]);
  return o;
};
import l from "react";
import i from "prop-types";
import p from "styled-components";
import E from "@strapi/icons/Dot";
import { NavLink as k } from "react-router-dom";
import { Box as a } from "../Box/Box.js";
import { Typography as b } from "../Typography/Typography.js";
import { Flex as $ } from "../Flex/Flex.js";
const x = p(a)`
  display: flex;
  align-items: center;
  justify-content: space-between;
  text-decoration: none;
  color: ${({ theme: e }) => e.colors.neutral800};
  svg > * {
    fill: ${({ theme: e }) => e.colors.neutral600};
  }

  &.active {
    ${({ theme: e }) => `
      background-color: ${e.colors.primary100};
      border-right: 2px solid ${e.colors.primary600};
      svg > * {
        fill: ${e.colors.primary700};
      }
      ${b} {
        color: ${e.colors.primary700};
        font-weight: 500;
      }
      `}
  }

  &:focus-visible {
    outline-offset: -2px;
  }
`, y = p(E)`
  width: ${12 / 16}rem;
  height: ${4 / 16}rem;
  * {
    fill: ${({ theme: e, $active: r }) => r ? e.colors.primary600 : e.colors.neutral600};
  }
`, S = p.div`
  svg {
    height: ${12 / 16}rem;
    width: ${12 / 16}rem;
  }
`, c = (w) => {
  var s = w, { children: e, icon: r, withBullet: o, isSubSectionChild: t } = s, h = g(s, ["children", "icon", "withBullet", "isSubSectionChild"]);
  return /* @__PURE__ */ l.createElement(x, f({
    as: k,
    icon: r,
    background: "neutral100",
    paddingLeft: t ? 9 : 7,
    paddingBottom: 2,
    paddingTop: 2
  }, h), /* @__PURE__ */ l.createElement($, null, r ? /* @__PURE__ */ l.createElement(S, null, r) : /* @__PURE__ */ l.createElement(y, null), /* @__PURE__ */ l.createElement(a, {
    paddingLeft: 2
  }, /* @__PURE__ */ l.createElement(b, {
    as: "span"
  }, e))), o && /* @__PURE__ */ l.createElement(a, {
    as: $,
    paddingRight: 4
  }, /* @__PURE__ */ l.createElement(y, {
    $active: !0
  })));
};
c.displayName = "SubNavLink";
c.defaultProps = {
  icon: null,
  isSubSectionChild: !1,
  withBullet: !1
};
c.propTypes = {
  children: i.node,
  icon: i.element,
  isSubSectionChild: i.bool,
  link: i.element,
  withBullet: i.bool
};
export {
  c as SubNavLink
};
