// @strapi/design-system/dist/Combobox/Combobox.js
var we = Object.defineProperty, $e = Object.defineProperties;
var ke = Object.getOwnPropertyDescriptors;
var w = Object.getOwnPropertySymbols;
var G = Object.prototype.hasOwnProperty, J = Object.prototype.propertyIsEnumerable;
var W = (o, i, a) => i in o ? we(o, i, { enumerable: !0, configurable: !0, writable: !0, value: a }) : o[i] = a, D = (o, i) => {
  for (var a in i || (i = {}))
    G.call(i, a) && W(o, a, i[a]);
  if (w)
    for (var a of w(i))
      J.call(i, a) && W(o, a, i[a]);
  return o;
}, F = (o, i) => $e(o, ke(i));
var Q = (o, i) => {
  var a = {};
  for (var p in o)
    G.call(o, p) && i.indexOf(p) < 0 && (a[p] = o[p]);
  if (o != null && w)
    for (var p of w(o))
      i.indexOf(p) < 0 && J.call(o, p) && (a[p] = o[p]);
  return a;
};
import r, { useState as O, useRef as y, useEffect as X, useLayoutEffect as Be, Children as Y, cloneElement as Re } from "react";
import e from "prop-types";
import { useId as Se } from "../helpers/useId.js";
import De from "@strapi/icons/CarretDown";
import Fe from "@strapi/icons/Cross";
import { filterOptions as Z, maintainScrollVisibility as Te, getActionFromKey as Ae, MenuActions as C, getUpdatedIndex as T } from "./utils.js";
import { Flex as ee } from "../Flex/Flex.js";
import { IconBox as Ve, CaretBox as Ke } from "../Select/components.js";
import { Popover as Ne } from "../Popover/Popover.js";
import { Box as Pe } from "../Box/Box.js";
import { Typography as te } from "../Typography/Typography.js";
import { Loader as qe } from "../Loader/Loader.js";
import { MainRow as He, InputContainer as Ue, ValueContainer as _e, Input as je } from "./components.js";
import { ComboboxOption as ze } from "./ComboboxOption.js";
import { ComboboxList as We } from "./ComboboxList.js";
import { Field as Ge } from "../Field/Field.js";
import { FieldLabel as Je } from "../Field/FieldLabel.js";
import "../Field/FieldInput.js";
import { FieldHint as Qe } from "../Field/FieldHint.js";
import { FieldError as Xe } from "../Field/FieldError.js";
import "../Field/FieldContext.js";
import "../Field/FieldAction.js";
import { Stack as Ye } from "../Stack/Stack.js";
import { KeyboardKeys as Ze } from "../helpers/keyboardKeys.js";
import { VisuallyHidden as et } from "../VisuallyHidden/VisuallyHidden.js";
const $ = (tt) => {
  var P = tt, {
    children: o,
    clearLabel: i,
    creatable: a,
    createMessage: p,
    disabled: f,
    error: k,
    hasMoreItems: re,
    hint: A,
    label: h,
    labelAction: ne,
    loading: B,
    loadingMessage: ae,
    noOptionsMessage: ie,
    onChange: V,
    onClear: K,
    onCreateOption: le,
    onInputChange: N,
    onLoadMore: se,
    placeholder: ce,
    required: pe,
    value: m
  } = P, ue = Q(P, [
    "children",
    "clearLabel",
    "creatable",
    "createMessage",
    "disabled",
    "error",
    "hasMoreItems",
    "hint",
    "label",
    "labelAction",
    "loading",
    "loadingMessage",
    "noOptionsMessage",
    "onChange",
    "onClear",
    "onCreateOption",
    "onInputChange",
    "onLoadMore",
    "placeholder",
    "required",
    "value"
  ]);
  const de = () => {
    var t;
    return (t = o.find((n) => {
      var l;
      return ((l = n.props) == null ? void 0 : l.value.toLowerCase()) === m.toLowerCase();
    }).props) == null ? void 0 : t.children;
  }, [c, q] = O(0), [fe, me] = O(null), [d, H] = O(o), [E, ge] = O(!1), [s, M] = O(""), x = y(), L = y(!1), I = y(), U = y(), be = y(), _ = y(!0), u = Se("combobox"), R = `${u}-label`;
  if (!h && !ue["aria-label"])
    throw new Error('The Combobox component needs a "label" or an "aria-label" props');
  X(() => {
    H(Z(o, s));
  }, [s, o]), X(() => {
    E && x.current && Te(x.current);
  }, [c]), Be(() => {
    if (_.current) {
      _.current = !1;
      return;
    }
  }, [m]);
  const Ce = E ? `${u}-${c}` : "", j = () => {
    V(null), M("");
  }, he = (t) => {
    N && N(t);
    const n = I.current.value;
    H(Z(o, n)), q(0), me(null), s !== n && M(n), E || g(!0, !1);
  }, Ee = (t) => {
    const { key: n } = t, l = a && s ? d.length : d.length - 1, v = Ae(n, E);
    switch (m && !s && n === Ze.BACKSPACE && j(), v) {
      case C.Next:
        return b(c === l ? 0 : T(c, l, v));
      case C.Previous:
        return b(c === 0 ? l : T(c, l, v));
      case C.Last:
      case C.First:
        return b(c === l ? 0 : T(c, l, v));
      case C.CloseSelect:
        t.preventDefault(), S(c);
        return;
      case C.Close:
        return t.preventDefault(), g(!1);
      case C.Open:
        return g(!0);
      default:
        return;
    }
  }, xe = (t) => {
    if (t.preventDefault(), m && !L.current && M(""), L.current) {
      L.current = !1;
      return;
    }
    g(!1, !1);
  }, b = (t) => {
    q(t);
  }, ve = (t) => {
    b(t), S(t);
  }, z = () => {
    L.current = !0;
  }, S = (t) => {
    const n = d[t];
    if (M(""), n)
      return V(n.props.value), g(!1);
    a && (le(s), g(!1));
  }, g = (t, n = !0) => {
    ge(t), n && I.current.focus();
  }, ye = Y.toArray(d).map((t, n) => {
    const l = c === n;
    return Re(t, {
      id: `${u}-${n}`,
      "aria-selected": fe === n,
      "aria-posinset": n + 1,
      "aria-setsize": Y.toArray(d).length,
      ref: (v) => {
        l && (x.current = v);
      },
      onClick: () => ve(n),
      onMouseDown: z,
      isSelected: l
    });
  }), Ie = () => {
    I.current.focus(), K && K(), j();
  }, Oe = () => {
    I.current.focus(), g(!0);
  }, Me = () => {
    const t = d.findIndex((n) => {
      var l;
      return ((l = n.props) == null ? void 0 : l.children) === s;
    });
    return s && t === -1;
  }, Le = (t) => {
    t.preventDefault(), g(t, !0);
  };
  return /* @__PURE__ */ r.createElement(Ge, {
    hint: A,
    error: k,
    id: u
  }, /* @__PURE__ */ r.createElement(et, {
    "aria-live": "polite",
    "aria-atomic": "false",
    "aria-relevant": "additions text"
  }, m), /* @__PURE__ */ r.createElement(Ye, {
    spacing: h || A || k ? 1 : 0
  }, h && /* @__PURE__ */ r.createElement(Je, {
    action: ne,
    required: pe,
    id: R
  }, h), /* @__PURE__ */ r.createElement(He, {
    ref: U,
    $disabled: f,
    hasError: k
  }, /* @__PURE__ */ r.createElement(Ue, {
    wrap: "wrap"
  }, !s && m && /* @__PURE__ */ r.createElement(_e, {
    id: `${u}-selected-value`
  }, /* @__PURE__ */ r.createElement(te, null, de())), /* @__PURE__ */ r.createElement(je, {
    "aria-activedescendant": Ce,
    "aria-autocomplete": "list",
    "aria-controls": `${u}-listbox`,
    "aria-disabled": f,
    "aria-expanded": E,
    "aria-haspopup": "listbox",
    "aria-labelledby": h ? R : void 0,
    id: u,
    onBlur: f ? void 0 : xe,
    onClick: f ? void 0 : Le,
    onInput: f ? void 0 : he,
    onKeyDown: f ? void 0 : Ee,
    placeholder: m ? "" : ce,
    readOnly: f,
    ref: I,
    role: "combobox",
    autoComplete: "off",
    autoCorrect: "off",
    spellCheck: "off",
    type: "text",
    value: s
  })), /* @__PURE__ */ r.createElement(ee, null, (m || s) && /* @__PURE__ */ r.createElement(Ve, {
    id: `${u}-clear`,
    "aria-label": i,
    disabled: f,
    paddingLeft: 3,
    as: "button",
    onClick: Ie,
    type: "button"
  }, /* @__PURE__ */ r.createElement(Fe, null)), /* @__PURE__ */ r.createElement(Ke, {
    disabled: f,
    paddingLeft: 3,
    "aria-hidden": !0,
    as: "button",
    onClick: Oe,
    tabIndex: -1,
    type: "button"
  }, /* @__PURE__ */ r.createElement(De, null)))), /* @__PURE__ */ r.createElement(Qe, null), /* @__PURE__ */ r.createElement(Xe, null)), E && /* @__PURE__ */ r.createElement(Ne, {
    id: `${u}-popover`,
    source: U,
    spacing: 4,
    fullWidth: !0,
    intersectionId: `${u}-listbox-popover-intersection`,
    onReachEnd: re && !B ? se : void 0
  }, /* @__PURE__ */ r.createElement("div", {
    role: "listbox",
    ref: be,
    id: `${u}-listbox`,
    "aria-labelledby": h ? R : void 0
  }, (Boolean(d.length) || a) && /* @__PURE__ */ r.createElement(r.Fragment, null, /* @__PURE__ */ r.createElement(We, {
    activeOptionRef: x,
    options: ye
  }), Me() && a && /* @__PURE__ */ r.createElement(ze, {
    isSelected: c === d.length,
    ref: (t) => {
      c === d.length && (x.current = t);
    },
    onMouseDown: z,
    onClick: () => S(),
    taindex: 0
  }, p(s))), !Boolean(d.length) && !a && !B && /* @__PURE__ */ r.createElement(Pe, {
    paddingLeft: 4,
    paddingRight: 4,
    paddingTop: 2,
    paddingBottom: 2,
    ref: x
  }, /* @__PURE__ */ r.createElement(te, {
    textColor: "neutral800"
  }, ie(s))), B && /* @__PURE__ */ r.createElement(ee, {
    justifyContent: "center",
    alignItems: "center",
    paddingTop: 2,
    paddingBottom: 2
  }, /* @__PURE__ */ r.createElement(qe, {
    small: !0
  }, ae)))));
}, oe = (o) => /* @__PURE__ */ r.createElement($, F(D({}, o), {
  creatable: !0
}));
$.defaultProps = oe.defaultProps = {
  "aria-label": void 0,
  clearLabel: "clear",
  creatable: !1,
  createMessage: (o) => `Create "${o}"`,
  disabled: !1,
  error: void 0,
  hasMoreItems: !1,
  hint: void 0,
  label: void 0,
  loading: !1,
  loadingMessage: "Loading content...",
  noOptionsMessage: () => "No results found",
  onClear: void 0,
  onCreateOption: void 0,
  onInputChange: void 0,
  onLoadMore: void 0,
  placeholder: "Select or enter a value",
  value: void 0
};
$.propTypes = {
  "aria-label": e.string,
  children: e.oneOfType([e.arrayOf(e.node), e.node]),
  clearLabel: e.string,
  creatable: e.bool,
  createMessage: e.func,
  disabled: e.bool,
  error: e.string,
  hasMoreItems: e.bool,
  hint: e.oneOfType([e.string, e.node, e.arrayOf(e.node)]),
  label: e.string,
  labelAction: e.element,
  loading: e.bool,
  loadingMessage: e.string,
  noOptionsMessage: e.func,
  onChange: e.func.isRequired,
  onClear: e.func,
  onCreateOption: e.func,
  onInputChange: e.func,
  onLoadMore: e.func,
  placeholder: e.string,
  value: e.string
};
oe.propTypes = F(D({}, $.propTypes), {
  onCreateOption: e.func.isRequired
});
export {
  $ as Combobox,
  oe as CreatableCombobox
};
