// @strapi/design-system/dist/Combobox/utils.js
import l from "compute-scroll-into-view";
import { KeyboardKeys as s } from "../helpers/keyboardKeys.js";
const t = {
  Close: "Close",
  CloseSelect: "CloseSelect",
  First: "First",
  Last: "Last",
  Next: "Next",
  Open: "Open",
  PageDown: "PageDown",
  PageUp: "PageUp",
  Previous: "Previous",
  Select: "Select",
  Space: "Space",
  Type: "Type"
}, f = {
  Close: "Close",
  First: "First",
  Last: "Last",
  Next: "Next",
  Open: "Open",
  Previous: "Previous",
  Select: "Select",
  UpLevel: "UpLevel"
};
function d(e = [], r, n = []) {
  const o = String(r != null ? r : "").toLowerCase();
  return o ? e.filter((i) => i.props.children.toString().toLowerCase().includes(o) && n.indexOf(i) < 0) : e;
}
function C(e, r) {
  if (!r && e === s.DOWN)
    return t.Open;
  if (e === s.DOWN)
    return t.Next;
  if (e === s.UP)
    return t.Previous;
  if (e === s.HOME)
    return t.First;
  if (e === s.END)
    return t.Last;
  if (e === s.ESCAPE)
    return t.Close;
  if (e === s.ENTER)
    return t.CloseSelect;
  if (e === s.BACKSPACE || e === s.CLEAR || e.length === 1)
    return t.Type;
}
function S(e, r, n) {
  switch (n) {
    case t.First:
      return 0;
    case t.Last:
      return r;
    case t.Previous:
      return Math.max(0, e - 1);
    case t.Next:
      return Math.min(r, e + 1);
    default:
      return e;
  }
}
function P(e) {
  l(e, {
    scrollMode: "if-needed",
    block: "nearest",
    inline: "nearest"
  }).forEach(({ el: n, top: o, left: i }) => {
    n.scrollTop = o, n.scrollLeft = i;
  });
}
export {
  t as MenuActions,
  f as TreeActions,
  d as filterOptions,
  C as getActionFromKey,
  S as getUpdatedIndex,
  P as maintainScrollVisibility
};
