// @strapi/design-system/dist/Flex/FlexProps.js
var a = Object.defineProperty;
var o = Object.getOwnPropertySymbols;
var l = Object.prototype.hasOwnProperty, m = Object.prototype.propertyIsEnumerable;
var s = (t, r, n) => r in t ? a(t, r, { enumerable: !0, configurable: !0, writable: !0, value: n }) : t[r] = n, i = (t, r) => {
  for (var n in r || (r = {}))
    l.call(r, n) && s(t, n, r[n]);
  if (o)
    for (var n of o(r))
      m.call(r, n) && s(t, n, r[n]);
  return t;
};
import f from "react";
import e from "prop-types";
const p = (t) => /* @__PURE__ */ f.createElement("div", i({}, t)), b = {
  alignItems: "center",
  basis: void 0,
  direction: "row",
  gap: void 0,
  inline: !1,
  justifyContent: void 0,
  reverse: !1,
  wrap: void 0
}, u = {
  alignItems: e.string,
  basis: e.oneOfType([e.string, e.number]),
  direction: e.string,
  gap: e.oneOfType([
    e.shape({
      desktop: e.number,
      mobile: e.number,
      tablet: e.number
    }),
    e.number,
    e.arrayOf(e.number),
    e.string
  ]),
  inline: e.bool,
  justifyContent: e.string,
  reverse: e.bool,
  shrink: e.number,
  wrap: e.string
};
p.defaultProps = b;
p.propTypes = u;
export {
  p as FlexProps,
  b as flexDefaultProps,
  u as flexPropTypes
};
