// @strapi/design-system/dist/Alert/Alert.cjs
"use strict";var b=Object.defineProperty;var u=Object.getOwnPropertySymbols;var x=Object.prototype.hasOwnProperty,y=Object.prototype.propertyIsEnumerable;var m=(e,a,t)=>a in e?b(e,a,{enumerable:!0,configurable:!0,writable:!0,value:t}):e[a]=t,d=(e,a)=>{for(var t in a||(a={}))x.call(a,t)&&m(e,t,a[t]);if(u)for(var t of u(a))y.call(a,t)&&m(e,t,a[t]);return e};var g=(e,a)=>{var t={};for(var l in e)x.call(e,l)&&a.indexOf(l)<0&&(t[l]=e[l]);if(e!=null&&u)for(var l of u(e))a.indexOf(l)<0&&y.call(e,l)&&(t[l]=e[l]);return t};Object.defineProperties(exports,{__esModule:{value:!0},[Symbol.toStringTag]:{value:"Module"}});const T=require("react"),$=require("prop-types"),k=require("styled-components"),R=require("@strapi/icons/Information"),A=require("@strapi/icons/CheckCircle"),I=require("@strapi/icons/ExclamationMarkCircle"),j=require("@strapi/icons/Cross"),s=require("../Box/Box.cjs"),C=require("../Typography/Typography.cjs"),w=require("../Flex/Flex.cjs"),i=require("./utils.cjs"),v=require("../themes/utils.cjs"),n=e=>e&&typeof e=="object"&&"default"in e?e:{default:e},r=n(T),o=n($),c=n(k),F=n(R),M=n(A),P=n(I),S=n(j),L=c.default(s.Box)`
  flex: 1;
  flex-wrap: wrap;
  display: flex;
  flex-direction: row;
`,W=c.default(s.Box)`
  border: 1px solid ${i.handleBorderColor};
  background: ${i.handleBackgroundColor};
  box-shadow: ${({theme:e})=>e.shadows.filterShadow};
`,z=c.default.button`
  border: none;
  background: transparent;
  font-size: ${12/16}rem;
  svg path {
    fill: ${({theme:e})=>e.colors.neutral700};
  }
  margin-top: ${({theme:e})=>e.spaces[1]};
  ${v.buttonFocusStyle};
`,O=c.default(s.Box)`
  font-size: ${20/16}rem;
  svg path {
    fill: ${i.handleIconColor};
  }
`,E=t=>{var l=t,{variant:e}=l,a=g(l,["variant"]);return e==="success"?r.default.createElement(M.default,d({},a)):e==="danger"?r.default.createElement(P.default,d({},a)):r.default.createElement(F.default,d({},a))},D=c.default(s.Box)`
  // Checked with the designers, validated
  padding-top: 1px;

  & a > span {
    color: ${i.handleIconColor};

    svg path {
      fill: ${i.handleIconColor};
    }
  }
`,f=G=>{var h=G,{title:e,children:a,variant:t,onClose:l,closeLabel:B,titleAs:_,action:p}=h,q=g(h,["title","children","variant","onClose","closeLabel","titleAs","action"]);return r.default.createElement(W,d({hasRadius:!0,paddingLeft:5,paddingRight:6,paddingTop:5,variant:t},q),r.default.createElement(w.Flex,{alignItems:"flex-start"},r.default.createElement(O,{paddingRight:3,variant:t},r.default.createElement(E,{variant:t,"aria-hidden":!0})),r.default.createElement(L,{role:t==="danger"?"alert":"status"},r.default.createElement(s.Box,{paddingBottom:2,paddingRight:1},r.default.createElement(C.Typography,{fontWeight:"bold",textColor:"neutral800",as:_},e)),r.default.createElement(s.Box,{paddingBottom:p?2:5,paddingRight:2},r.default.createElement(C.Typography,{as:"p",textColor:"neutral800"},a)),p&&r.default.createElement(D,{paddingBottom:5,variant:t},p)),r.default.createElement(z,{onClick:l,"aria-label":B},r.default.createElement(S.default,{"aria-hidden":!0}))))};f.defaultProps={action:void 0,variant:"default",titleAs:"p"};f.propTypes={action:o.default.element,children:o.default.node.isRequired,closeLabel:o.default.string.isRequired,onClose:o.default.func.isRequired,title:o.default.string.isRequired,titleAs:o.default.string,variant:o.default.oneOf(["danger","success","default"])};E.propTypes={variant:f.propTypes.variant};exports.Alert=f;
