// @strapi/design-system/dist/RawTable/RawTable.js
var q = Object.defineProperty;
var f = Object.getOwnPropertySymbols;
var p = Object.prototype.hasOwnProperty, E = Object.prototype.propertyIsEnumerable;
var y = (t, a, r) => a in t ? q(t, a, { enumerable: !0, configurable: !0, writable: !0, value: r }) : t[a] = r, P = (t, a) => {
  for (var r in a || (a = {}))
    p.call(a, r) && y(t, r, a[r]);
  if (f)
    for (var r of f(a))
      E.call(a, r) && y(t, r, a[r]);
  return t;
};
var T = (t, a) => {
  var r = {};
  for (var l in t)
    p.call(t, l) && a.indexOf(l) < 0 && (r[l] = t[l]);
  if (t != null && f)
    for (var l of f(t))
      a.indexOf(l) < 0 && E.call(t, l) && (r[l] = t[l]);
  return r;
};
import K, { useRef as R, useState as d, useCallback as w, useEffect as A } from "react";
import i from "prop-types";
import { RawTableContext as F } from "./RawTableContext.js";
import { KeyboardKeys as s } from "../helpers/keyboardKeys.js";
import { focusFocusable as H } from "./focusFocusable.js";
const _ = (U) => {
  var b = U, { colCount: t, rowCount: a, jumpStep: r, initialCol: l, initialRow: G } = b, N = T(b, ["colCount", "rowCount", "jumpStep", "initialCol", "initialRow"]);
  const m = R(null), o = R(!1), [D, n] = d(G), [k, u] = d(l), O = w(({ colIndex: c, rowIndex: e }) => {
    u(c), n(e);
  }, []);
  A(() => {
    o.current && H(m.current), o.current || (o.current = !0);
  }, [k, D]);
  const h = (c) => {
    switch (c.key) {
      case s.RIGHT: {
        c.preventDefault(), u((e) => e < t - 1 ? e + 1 : e);
        break;
      }
      case s.LEFT: {
        c.preventDefault(), u((e) => e > 0 ? e - 1 : e);
        break;
      }
      case s.UP: {
        c.preventDefault(), n((e) => e > 0 ? e - 1 : e);
        break;
      }
      case s.DOWN: {
        c.preventDefault(), n((e) => e < a - 1 ? e + 1 : e);
        break;
      }
      case s.HOME: {
        c.preventDefault(), c.ctrlKey && n(0), u(0);
        break;
      }
      case s.END: {
        c.preventDefault(), c.ctrlKey && n(a - 1), u(t - 1);
        break;
      }
      case s.PAGE_DOWN: {
        c.preventDefault(), n((e) => e + r < a ? e + r : a - 1);
        break;
      }
      case s.PAGE_UP: {
        c.preventDefault(), n((e) => e - r > 0 ? e - r : 0);
        break;
      }
    }
  };
  return /* @__PURE__ */ K.createElement(F.Provider, {
    value: { rowIndex: D, colIndex: k, setTableValues: O }
  }, /* @__PURE__ */ K.createElement("table", P({
    ref: m,
    "aria-rowcount": a,
    "aria-colcount": t,
    onKeyDown: h
  }, N)));
};
_.defaultProps = {
  jumpStep: 3,
  initialCol: 0,
  initialRow: 0
};
_.propTypes = {
  colCount: i.number.isRequired,
  initialCol: i.number,
  initialRow: i.number,
  jumpStep: i.number,
  rowCount: i.number.isRequired
};
export {
  _ as RawTable
};
