// @strapi/design-system/dist/Select/Select.js
var ie = Object.defineProperty;
var I = Object.getOwnPropertySymbols;
var q = Object.prototype.hasOwnProperty, z = Object.prototype.propertyIsEnumerable;
var P = (o, l, i) => l in o ? ie(o, l, { enumerable: !0, configurable: !0, writable: !0, value: i }) : o[l] = i, G = (o, l) => {
  for (var i in l || (l = {}))
    q.call(l, i) && P(o, i, l[i]);
  if (I)
    for (var i of I(l))
      z.call(l, i) && P(o, i, l[i]);
  return o;
};
var H = (o, l) => {
  var i = {};
  for (var a in o)
    q.call(o, a) && l.indexOf(a) < 0 && (i[a] = o[a]);
  if (o != null && I)
    for (var a of I(o))
      l.indexOf(a) < 0 && z.call(o, a) && (i[a] = o[a]);
  return i;
};
import r, { useState as le, useRef as ae, Children as W, cloneElement as _ } from "react";
import e from "prop-types";
import se from "@strapi/icons/CarretDown";
import pe from "@strapi/icons/Cross";
import { sizes as ce } from "../themes/sizes.js";
import { SelectButton as me } from "./SelectButton.js";
import { Field as de } from "../Field/Field.js";
import { FieldLabel as fe } from "../Field/FieldLabel.js";
import "../Field/FieldInput.js";
import { FieldHint as ue } from "../Field/FieldHint.js";
import { FieldError as he } from "../Field/FieldError.js";
import "../Field/FieldContext.js";
import "../Field/FieldAction.js";
import { Popover as Ee } from "../Popover/Popover.js";
import { Stack as be } from "../Stack/Stack.js";
import { Typography as N } from "../Typography/Typography.js";
import { Flex as $ } from "../Flex/Flex.js";
import { Box as J } from "../Box/Box.js";
import { useId as ye } from "../helpers/useId.js";
import { SelectList as ge } from "./SelectList.js";
import { SelectButtonWrapper as ve, IconBox as Ie, CaretBox as Oe } from "./components.js";
import { useButtonRef as Ce } from "./hooks/useButtonRef.js";
import { VisuallyHidden as K } from "../VisuallyHidden/VisuallyHidden.js";
import { DownState as Se } from "./constants.js";
import { escapeSelector as Q } from "../helpers/escapeSelector.js";
import { SelectTags as xe } from "./SelectTags.js";
import Be from "styled-components";
const Te = Be($)`
  width: 100%;
`, U = ($e) => {
  var w = $e, {
    label: o,
    labelAction: l,
    id: i,
    children: a,
    customizeContent: h,
    placeholder: O,
    onChange: C,
    value: n,
    hint: S,
    error: u,
    disabled: m,
    clearLabel: X,
    onClear: R,
    onReachEnd: Y,
    multi: p,
    required: Z,
    size: V,
    startIcon: L,
    withTags: E
  } = w, ee = H(w, [
    "label",
    "labelAction",
    "id",
    "children",
    "customizeContent",
    "placeholder",
    "onChange",
    "value",
    "hint",
    "error",
    "disabled",
    "clearLabel",
    "onClear",
    "onReachEnd",
    "multi",
    "required",
    "size",
    "startIcon",
    "withTags"
  ]);
  const s = ye("select", i), [b, y] = le(void 0), D = Ce(b), F = ae(null), x = `${s}-label`, g = `${s}-content`, te = u ? `${s}-error` : S ? `${s}-hint` : void 0;
  if (E && !p)
    throw new Error('The "withTags" props can only be used when the "multi" prop is present');
  const re = () => {
    y(void 0);
  }, oe = () => {
    m || (R(), D.current.focus());
  }, M = (t) => {
    t.preventDefault(), !m && (t.nativeEvent.which === 3 || t.nativeEvent.button === 2 || y(Se.Mouse));
  }, B = (t, c = !0) => {
    p ? C(n.includes(t) ? n.filter((f) => f !== t) : [...n, t]) : (C(t), c && y(void 0));
  }, k = (t) => {
    C(
      n.includes(t[0]) ? n.filter(function(c) {
        return this.indexOf(c) < 0;
      }, t) : [...n, ...t]
    );
  };
  let T, j = [];
  const A = (t, c) => {
    const f = `${s}-option-${t.props.value}`, v = p ? n.includes(t.props.value) : t.props.value === n;
    return v && (E ? j.push({ label: t.props.children, value: t.props.value }) : T = t.props.children), _(t, {
      id: Q(f),
      onClick: () => B(t.props.value),
      selected: v,
      multi: p,
      isChild: c
    });
  }, ne = W.toArray(a).map((t) => {
    if (t.type.displayName === "OptGroup") {
      const c = `${s}-option-${t.props.label}`, f = t.props.children.every((d) => n.includes(d.props.value)), v = !f && t.props.children.some((d) => n.includes(d.props.value));
      return _(t, {
        id: Q(c),
        onClick: () => k(t.props.children.map((d) => d.props.value)),
        selected: f,
        indeterminate: v,
        multi: p,
        children: W.toArray(t.props.children).map((d) => A(d, !0)),
        value: t.props.label
      });
    } else
      return A(t);
  });
  return /* @__PURE__ */ r.createElement(de, {
    hint: S,
    error: u,
    id: s
  }, /* @__PURE__ */ r.createElement(be, {
    spacing: o || S || u ? 1 : 0
  }, o && /* @__PURE__ */ r.createElement(fe, {
    required: Z,
    as: "span",
    id: x,
    action: l
  }, o), /* @__PURE__ */ r.createElement(ve, {
    size: V,
    hasError: Boolean(u),
    disabled: m,
    ref: F
  }, /* @__PURE__ */ r.createElement(me, G({
    ref: D,
    labelledBy: `${x} ${g}`,
    "aria-describedby": te,
    expanded: Boolean(b),
    onTrigger: y,
    id: s,
    hasError: Boolean(u),
    disabled: m,
    onMouseDown: M
  }, ee)), /* @__PURE__ */ r.createElement(Te, {
    justifyContent: "space-between"
  }, /* @__PURE__ */ r.createElement($, null, L && /* @__PURE__ */ r.createElement(J, {
    paddingLeft: 3,
    "aria-hidden": !0
  }, L), E && /* @__PURE__ */ r.createElement(xe, {
    tags: j,
    onRemoveTag: B,
    disabled: m
  }), /* @__PURE__ */ r.createElement(J, {
    paddingLeft: 4,
    paddingRight: 4
  }, E ? /* @__PURE__ */ r.createElement(r.Fragment, null, !n || n.length === 0 ? /* @__PURE__ */ r.createElement(N, {
    ellipsis: !0,
    id: g,
    textColor: "neutral600"
  }, O) : null, /* @__PURE__ */ r.createElement(K, {
    as: "span",
    id: g
  }, h ? h(n) : T || O, n.join(", "))) : /* @__PURE__ */ r.createElement(N, {
    ellipsis: !0,
    id: g,
    textColor: n ? "neutral800" : "neutral600"
  }, h ? h(n) : T || O, p && /* @__PURE__ */ r.createElement(K, {
    as: "span"
  }, n.join(", "))))), /* @__PURE__ */ r.createElement($, null, (p && n && n.length || !p && n) && R && /* @__PURE__ */ r.createElement(Ie, {
    as: "button",
    type: "button",
    onClick: oe,
    "aria-label": X,
    "aria-disabled": m
  }, /* @__PURE__ */ r.createElement(pe, null)), /* @__PURE__ */ r.createElement(Oe, {
    paddingLeft: 3,
    "aria-hidden": !0,
    as: "button",
    type: "button",
    onMouseDown: M,
    tabIndex: -1,
    disabled: m
  }, /* @__PURE__ */ r.createElement(se, null))))), /* @__PURE__ */ r.createElement(ue, null), /* @__PURE__ */ r.createElement(he, null)), b && /* @__PURE__ */ r.createElement(Ee, {
    source: F,
    spacing: 4,
    fullWidth: !0,
    intersectionId: `select-list-intersection-${s}`,
    onReachEnd: Y
  }, /* @__PURE__ */ r.createElement(ge, {
    selectId: s,
    labelledBy: o ? x : void 0,
    onEscape: re,
    expanded: b,
    onSelectItem: (t, c) => c ? k(t) : B(t, !1),
    multi: p
  }, ne)));
};
U.defaultProps = {
  "aria-label": void 0,
  children: [],
  clearLabel: "Clear",
  customizeContent: void 0,
  disabled: !1,
  id: void 0,
  label: void 0,
  labelAction: void 0,
  multi: !1,
  onChange: () => {
  },
  onClear: void 0,
  onReachEnd: void 0,
  value: void 0,
  hint: void 0,
  error: void 0,
  placeholder: "Select...",
  required: !1,
  size: "M",
  startIcon: void 0,
  withTags: !1
};
U.propTypes = {
  "aria-label": e.string,
  children: e.oneOfType([e.arrayOf(e.node), e.node]),
  clearLabel: e.string,
  customizeContent: e.func,
  disabled: e.bool,
  error: e.string,
  hint: e.oneOfType([e.string, e.node, e.arrayOf(e.node)]),
  id: e.oneOfType([e.string, e.number]),
  label: e.string,
  labelAction: e.element,
  multi: e.bool,
  onChange: e.func,
  onClear: e.func,
  onReachEnd: e.func,
  placeholder: e.string,
  required: e.bool,
  size: e.oneOf(Object.keys(ce.input)),
  startIcon: e.element,
  value: e.oneOfType([
    e.arrayOf(e.oneOfType([e.string, e.number])),
    e.string,
    e.number
  ]),
  withTags: e.bool
};
export {
  U as Select
};
