// @strapi/design-system/dist/v2/SubNav/SubNavLink.cjs
"use strict";var k=Object.defineProperty;var n=Object.getOwnPropertySymbols;var f=Object.prototype.hasOwnProperty,y=Object.prototype.propertyIsEnumerable;var p=(e,t,l)=>t in e?k(e,t,{enumerable:!0,configurable:!0,writable:!0,value:l}):e[t]=l,g=(e,t)=>{for(var l in t||(t={}))f.call(t,l)&&p(e,l,t[l]);if(n)for(var l of n(t))y.call(t,l)&&p(e,l,t[l]);return e};var m=(e,t)=>{var l={};for(var o in e)f.call(e,o)&&t.indexOf(o)<0&&(l[o]=e[o]);if(e!=null&&n)for(var o of n(e))t.indexOf(o)<0&&y.call(e,o)&&(l[o]=e[o]);return l};Object.defineProperties(exports,{__esModule:{value:!0},[Symbol.toStringTag]:{value:"Module"}});const L=require("react"),T=require("prop-types"),_=require("styled-components"),S=require("@strapi/icons/Dot"),u=require("../../Box/Box.cjs"),x=require("../../Typography/Typography.cjs"),b=require("../../Flex/Flex.cjs"),q=require("../../BaseLink/BaseLink.cjs"),s=e=>e&&typeof e=="object"&&"default"in e?e:{default:e},a=s(L),r=s(T),c=s(_),E=s(S),j=c.default(u.Box)`
  display: flex;
  align-items: center;
  justify-content: space-between;
  text-decoration: none;
  color: ${({theme:e})=>e.colors.neutral800};
  svg > * {
    fill: ${({theme:e})=>e.colors.neutral600};
  }

  &.active {
    ${({theme:e})=>`
      background-color: ${e.colors.primary100};
      border-right: 2px solid ${e.colors.primary600};
      svg > * {
        fill: ${e.colors.primary700};
      }
      ${x.Typography} {
        color: ${e.colors.primary700};
        font-weight: 500;
      }
      `}
  }

  &:focus-visible {
    outline-offset: -2px;
  }
`,h=c.default(E.default)`
  width: ${12/16}rem;
  height: ${4/16}rem;
  * {
    fill: ${({theme:e,$active:t})=>t?e.colors.primary600:e.colors.neutral600};
  }
`,w=c.default.div`
  svg {
    height: ${12/16}rem;
    width: ${12/16}rem;
  }
`,i=a.default.forwardRef((F,v)=>{var d=F,{children:e,icon:t,withBullet:l,as:o,isSubSectionChild:B}=d,$=m(d,["children","icon","withBullet","as","isSubSectionChild"]);return a.default.createElement(j,g({as:o,icon:t,background:"neutral100",paddingLeft:B?9:7,paddingBottom:2,paddingTop:2,ref:v},$),a.default.createElement(b.Flex,null,t?a.default.createElement(w,null,t):a.default.createElement(h,null),a.default.createElement(u.Box,{paddingLeft:2},a.default.createElement(x.Typography,{as:"span"},e))),l&&a.default.createElement(u.Box,{as:b.Flex,paddingRight:4},a.default.createElement(h,{$active:!0})))});i.displayName="SubNavLink";i.defaultProps={as:q.BaseLink,icon:null,isSubSectionChild:!1,withBullet:!1};i.propTypes={as:r.default.elementType,children:r.default.node,icon:r.default.element,isSubSectionChild:r.default.bool,link:r.default.element,withBullet:r.default.bool};exports.SubNavLink=i;
